flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/MetricUtil.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.connector.kafka;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.metrics.Counter;
+
+import org.apache.kafka.common.Metric;
+import org.apache.kafka.common.MetricName;
+
+import java.util.Map;
+
+/** Collection of methods to interact with Kafka's client metric system. */
+@Internal
+public class MetricUtil {
+
+    /**
+     * Tries to find the the Kafka {@link Metric} in the provided metrics.
+     *
+     * @return {@link Metric} which exposes continuous updates
+     * @throws IllegalStateException if the metric is not part of the provided metrics
+     */
+    public static Metric getKafkaMetric(
+            Map<MetricName, ? extends Metric> metrics, String metricGroup, String metricName) {
```

Comment:
What about having another ```Predicate<Metric>``` as an argument? There might be multiple metrics with the same group and metric name but different in tags, such as "records-lag". The predicate can let user to filter the specific metric they want.

Reply:
That sounds like a good idea 👍  I've added another util to filter with a given Predicate.